#! /bin/sh -e
## 01_multiarch.dpatch by Peter De Schrijver <p2@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix the configure script to allow multiple builds from the same 
## DP: sourcetree

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -ur orig/alsa-lib-1.0.10+1.0.11rc1/configure.in alsa-lib-1.0.10+1.0.11rc1/configure.in
--- orig/alsa-lib-1.0.10+1.0.11rc1/configure.in	2005-12-13 08:51:52.000000000 +0100
+++ alsa-lib-1.0.10+1.0.11rc1/configure.in	2006-01-16 21:35:11.000000000 +0100
@@ -328,13 +328,6 @@
 fi
 
 
-dnl Create PCM plugin symbol list for static library
-rm -f src/pcm/pcm_symbols_list.c
-for t in $PCM_PLUGIN_LIST; do
-  if eval test \$build_pcm_$t = yes; then
-    echo \&_snd_module_pcm_$t, >> src/pcm/pcm_symbols_list.c
-  fi
-done
 
 dnl Make a symlink for inclusion of alsa/xxx.h
 if test ! -L include/alsa ; then
@@ -355,3 +348,11 @@
 	  modules/Makefile modules/mixer/Makefile modules/mixer/simple/Makefile \
 	  alsalisp/Makefile aserver/Makefile test/Makefile utils/Makefile \
           utils/alsa-lib.spec utils/alsa.pc)
+
+dnl Create PCM plugin symbol list for static library
+rm -f src/pcm/pcm_symbols_list.c
+for t in $PCM_PLUGIN_LIST; do
+  if eval test \$build_pcm_$t = yes; then
+    echo \&_snd_module_pcm_$t, >> src/pcm/pcm_symbols_list.c
+  fi
+done
